Struggling with React 16/Office UI Fabric React in SPFX

#### Category
- [X] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

I would like to use the React 16/Office UI Fabric with SPFX but struggling with getting SPFX to build.

#### Observed Behavior

Cannot build the solution.

The documentation page for [https://docs.microsoft.com/en-us/sharepoint/dev/spfx/office-ui-fabric-integration](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/office-ui-fabric-integration) is quite far out of date. Can this be updated?

Also, this thread indicated that typescript isn't up to date to support the latest React 16/Office UI Fabric:
[https://github.com/OfficeDev/office-ui-fabric-react/issues/7054](https://github.com/OfficeDev/office-ui-fabric-react/issues/7054)

I have tried the suggestion and does not work.

[14:17:55] Using gulpfile C:\git\tfs\intranet-web-parts\alerts\gulpfile.js
[14:17:55] Starting gulp
[14:17:55] Starting 'build'...
[14:17:55] Starting subtask 'configure-sp-build-rig'...
[14:17:55] Finished subtask 'configure-sp-build-rig' after 5.71 ms
[14:17:55] Starting subtask 'pre-copy'...
[14:17:55] Finished subtask 'pre-copy' after 49 ms
[14:17:55] Starting subtask 'copy-static-assets'...
[14:17:55] Starting subtask 'sass'...
[14:17:55] Finished subtask 'copy-static-assets' after 148 ms
[14:17:55] Finished subtask 'sass' after 231 ms
[14:17:55] Starting subtask 'tslint'...
[14:17:55] [tslint] tslint version: 5.9.1
[14:17:55] Starting subtask 'tsc'...
[14:17:55] [tsc] typescript version: 2.4.2
[14:17:58] Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,67): error TS1005: ';' expected.
[14:17:58] Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,114): error TS1005: '(' expected.
[14:17:58] Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,136): error TS1005: '(' expected.
[14:17:58] Error - 'tsc' sub task errored after 2.4 s
 exited with code 2
[14:17:58] 'build' errored after 2.91 s
[14:17:58]
[14:17:58] ==================[ Finished ]==================
Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,67): error TS1005: ';' expected.
Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,114): error TS1005: '(' expected.
Error - [tsc] node_modules/office-ui-fabric-react/node_modules/@uifabric/merge-styles/lib/IStyleSet.d.ts(15,136): error TS1005: '(' expected.
Error - 'tsc' sub task errored after 2.4 s
 exited with code 2
[14:17:59] Finished subtask 'tslint' after 3.63 s
[14:17:59] Project spfx-alerts version:0.0.1
[14:17:59] Build tools version:3.8.33
[14:17:59] Node version:v8.11.4
[14:17:59] Total duration:7.81 s
[14:17:59] Task errors:4

#### Steps to Reproduce

Setup SPFX with the following dependencies:

- Setup SPFX Web Part with React
- Remove @microsoft/sp-office-ui-fabric-core@1.7.0
- Use dependencies:
- - office-ui-fabric-react@6.57.0

  "dependencies": {
    "@microsoft/sp-core-library": "1.7.0",
    "@microsoft/sp-lodash-subset": "1.7.0",
    "@microsoft/sp-webpart-base": "1.7.0",
    "@types/es6-promise": "0.0.33",
    "@types/react": "16.4.2",
    "@types/react-dom": "16.0.5",
    "@types/webpack-env": "1.13.1",
    "office-ui-fabric-react": "6.57.0",
    "react": "16.3.2",
    "react-dom": "16.3.2"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "1.7.0",
    "@microsoft/sp-tslint-rules": "1.7.0",
    "@microsoft/sp-module-interfaces": "1.7.0",
    "@microsoft/sp-webpart-workbench": "1.7.0",
    "gulp": "~3.9.1",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2"
  }

Please can you confirm what versions of React/Typescript/Office UI Fabric to use? Or any additional dependencies to install.

Thanks for your contribution! Sharing is caring.

--- a/docs/spfx/office-ui-fabric-integration.md
+++ b/docs/spfx/office-ui-fabric-integration.md
@@ -85,7 +85,7 @@ Now you can use the core styles as mixins and variables.
 
 ## Office UI Fabric React
 
-We recommend that you install the latest Office UI Fabric React package and place an explicit dependency on that specific version of the package. This includes the components used in your SharePoint Framework solution in your component bundle, either the web part or extension depending on where you use the Fabric React components. 
+We recommend that you use the versions of the Office UI Fabric React package included in the project in the SharePoint Framework's Yeoman genereator. For instance, the SharePoint Framework v1.7.0 release uses Fabric React v5.131.0
 
 > [!NOTE] 
 > Fabric React versions 2.x or older are not supported in SharePoint Framework.

